### DISCUSSION TITLE
Not sure how to write a locally ambiguous rule

### DISCUSSION THREAD

**Discussion by mr-martian** [tree-sitter/tree-sitter#1820] [Q&A] [answered]
So I'm dealing with the following:
```
X =>      (pattern_token (lexicon_identifier X))
X? =>     (pattern_token (lexicon_identifier X) (operator ?))
X?(1) =>  (pattern_token (lexicon_identifer X ? (1) ))
X(1) =>   (pattern_token (lexicon_identifer X (1) ))
X(1)? =>  (pattern_token (lexicon_identifier X (1) ) (operator ?))
X?(1)? => (pattern_token (lexicon_identifier X ? (1) ) (operator ?))
```
basically, I have an identifier, and if it's followed by `?`, the role of that character is determined by whether the immediately following character is an open paren.

Everything I've tried so far has either given a rule conflict error or else has failed to parse `X?`. Any ideas?

**Comment by ahelwer** [comment, marked as answer]
What's wrong with declaring a conflict and using GLR parsing?

**Reply by mr-martian** [reply]
I guess I misread the docs on how to do that the first couple times through.

Now I just have to deal with the oddity that

```js
    pattern_line: $ => seq(
      $.pattern_item,
      repeat(seq($._ws, $.pattern_item)),
      $._nl
    ),
    pattern_item: $ => choice(
      $.pattern_token,
      $.pattern_option,
      $.left_sieve,
      $.right_sieve
    ),
```
isn't working, but if I replace references to `$.pattern_item` with the `choice()` itself it works just fine, but that's only 2 rules so whatever.
